mydumper.c

```diff
@@ -895,14 +901,67 @@ MYSQL *reconnect_for_binlog(MYSQL *thrconn) {
 	mysql_options(thrconn, MYSQL_OPT_READ_TIMEOUT, (const char*)&timeout);
 
 
-	if (!mysql_real_connect(thrconn, hostname, username, password, NULL, port, socket_path, 0)) {
+	if (!mysql_connect_wrap(thrconn, username, password, NULL, port, socket_path, 0)) {
 		g_critical("Failed to re-connect to database: %s", mysql_error(thrconn));
 		exit(EXIT_FAILURE);
 	}
 	return thrconn;
 }
 #endif
 
+void pre_resolve_host(const char *host) {
+	if (host == NULL) {
+		return;
+	}
+
+	struct addrinfo hints, *res;
+	int errcode;
+	char addrstr[100];
+	void *ptr;
+
+	memset(&hints, 0, sizeof(hints));
+	hints.ai_family = PF_UNSPEC;
+	hints.ai_socktype = SOCK_STREAM;
+
+	errcode = getaddrinfo(host, NULL, &hints, &res);
+	if (errcode != 0) {
+ 		g_print("failed to resolve host %s\n", host);
+		exit(EXIT_FAILURE);
+	}
+
+	while (res) {
+		if (res->ai_family == AF_INET6) {
+			ptr = &((struct sockaddr_in6 *) res->ai_addr)->sin6_addr;
+		} else {
```

Comment:
```suggestion
		} else if (res->ai_family == AF_INET) {
```

and skip the result (or exit) if the family is neither.